### PR TITLE
vcsim: workaround xml ns issue with pyvsphere

### LINKED ISSUE #945: vcsim creates stack trace
vcsim creates following stack trace when pyvsphere library is used to get details about datacenter - 

```
2017/12/02 14:35:45 http: panic serving 127.0.0.1:55365: runtime error: invalid memory address or nil pointer dereference
goroutine 1422 [running]:
net/http.(*conn).serve.func1(0xc4203225a0)
	/usr/local/Cellar/go/1.9.1/libexec/src/net/http/server.go:1697 +0xd0
panic(0x180ee60, 0x2082fc0)
	/usr/local/Cellar/go/1.9.1/libexec/src/runtime/panic.go:491 +0x283
github.com/vmware/govmomi/simulator.(*retrieveResult).selectSet(0xc4204cb3f8, 0x196d6e0, 0xc420088b60, 0x199, 0xc4201b6c00, 0x9, 0x10, 0xc4204cb3e0, 0x0, 0x18f5e00)
	/Volumes/data/src/go/src/github.com/vmware/govmomi/simulator/property_collector.go:325 +0x4b7
github.com/vmware/govmomi/simulator.(*PropertyCollector).collect(0xc420060800, 0xc42013e2c0, 0x0, 0x2, 0x1002e08)
	/Volumes/data/src/go/src/github.com/vmware/govmomi/simulator/property_collector.go:376 +0x454
github.com/vmware/govmomi/simulator.(*PropertyCollector).RetrievePropertiesEx(0xc420060800, 0xc42013e2c0, 0x0, 0x0)
	/Volumes/data/src/go/src/github.com/vmware/govmomi/simulator/property_collector.go:435 +0x53
reflect.Value.call(0x1980d60, 0xc420060800, 0x1a13, 0x19d58cd, 0x4, 0xc4204cba70, 0x1, 0x1, 0x1980d60, 0x1782ac0, ...)
	/usr/local/Cellar/go/1.9.1/libexec/src/reflect/value.go:434 +0x906
reflect.Value.Call(0x1980d60, 0xc420060800, 0x1a13, 0xc4204cba70, 0x1, 0x1, 0xc420060800, 0x1a13, 0xc4201f6cc0)
	/usr/local/Cellar/go/1.9.1/libexec/src/reflect/value.go:302 +0xa4
github.com/vmware/govmomi/simulator.(*Service).call(0xc4200e46e0, 0xc420345500, 0xe00, 0xc420345500)
	/Volumes/data/src/go/src/github.com/vmware/govmomi/simulator/simulator.go:144 +0x5b0
github.com/vmware/govmomi/simulator.(*Service).ServeSDK(0xc4200e46e0, 0x2057320, 0xc4202da0e0, 0xc4201b6800)
	/Volumes/data/src/go/src/github.com/vmware/govmomi/simulator/simulator.go:272 +0xbe2
github.com/vmware/govmomi/simulator.(*Service).ServeSDK-fm(0x2057320, 0xc4202da0e0, 0xc4201b6800)
	/Volumes/data/src/go/src/github.com/vmware/govmomi/simulator/simulator.go:414 +0x48
net/http.HandlerFunc.ServeHTTP(0xc420282280, 0x2057320, 0xc4202da0e0, 0xc4201b6800)
	/usr/local/Cellar/go/1.9.1/libexec/src/net/http/server.go:1918 +0x44
net/http.(*ServeMux).ServeHTTP(0xc420288780, 0x2057320, 0xc4202da0e0, 0xc4201b6800)
	/usr/local/Cellar/go/1.9.1/libexec/src/net/http/server.go:2254 +0x130
net/http.serverHandler.ServeHTTP(0xc4202865b0, 0x2057320, 0xc4202da0e0, 0xc4201b6800)
	/usr/local/Cellar/go/1.9.1/libexec/src/net/http/server.go:2619 +0xb4
net/http.(*conn).serve(0xc4203225a0, 0x20579a0, 0xc4203c2e40)
	/usr/local/Cellar/go/1.9.1/libexec/src/net/http/server.go:1801 +0x71d
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.9.1/libexec/src/net/http/server.go:2720 +0x288
```

Python stack trace is 

```
The full traceback is:
Traceback (most recent call last):
  File "/var/folders/2x/p__2sw2x0qv4581wwp5g45w00000gp/T/ansible_SrU1cj/ansible_module_vsphere_guest.py", line 1937, in <module>
    main()
  File "/var/folders/2x/p__2sw2x0qv4581wwp5g45w00000gp/T/ansible_SrU1cj/ansible_module_vsphere_guest.py", line 1927, in main
    state=state
  File "/var/folders/2x/p__2sw2x0qv4581wwp5g45w00000gp/T/ansible_SrU1cj/ansible_module_vsphere_guest.py", line 1220, in create_vm
    v in vsphere_client.get_datacenters().items() if v == datacenter]
  File "/Volumes/data/src/ansible/venv4/lib/python2.7/site-packages/pysphere/vi_server.py", line 208, in get_datacenters
    return self._get_managed_objects(MORTypes.Datacenter, from_mor)
  File "/Volumes/data/src/ansible/venv4/lib/python2.7/site-packages/pysphere/vi_server.py", line 759, in _get_managed_objects
    obj_type=mo_type)
  File "/Volumes/data/src/ansible/venv4/lib/python2.7/site-packages/pysphere/vi_server.py", line 684, in _retrieve_properties_traversal
    return request_call(request)
  File "/Volumes/data/src/ansible/venv4/lib/python2.7/site-packages/pysphere/vi_server.py", line 700, in call_retrieve_properties_ex
    request)._returnval
  File "/Volumes/data/src/ansible/venv4/lib/python2.7/site-packages/pysphere/resources/VimService_services.py", line 115, in RetrievePropertiesEx
    response = self.binding.Receive(RetrievePropertiesExResponseMsg.typecode)
  File "/Volumes/data/src/ansible/venv4/lib/python2.7/site-packages/pysphere/ZSI/client.py", line 540, in Receive
    self.ReceiveSOAP(**kw)
  File "/Volumes/data/src/ansible/venv4/lib/python2.7/site-packages/pysphere/ZSI/client.py", line 423, in ReceiveSOAP
    if not self.IsSOAP():
  File "/Volumes/data/src/ansible/venv4/lib/python2.7/site-packages/pysphere/ZSI/client.py", line 415, in IsSOAP
    self.ReceiveRaw()
  File "/Volumes/data/src/ansible/venv4/lib/python2.7/site-packages/pysphere/ZSI/client.py", line 382, in ReceiveRaw
    response = self.local.h.getresponse()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 1132, in getresponse
    response.begin()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 453, in begin
    version, status, reason = self._read_status()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 417, in _read_status
    raise BadStatusLine(line)
httplib.BadStatusLine: ''

```

### PR DESCRIPTION
**Pull request by dougm**
Fixes #945

### CODE REVIEW

File: simulator/simulator.go [hunk 1]
```diff
@@ -491,7 +491,23 @@ func (s *Server) Close() {
...
+var (
+	vim25MapType = types.TypeFunc()
+	typeFunc     = defaultMapType
+)
+
+func defaultMapType(name string) (reflect.Type, bool) {
+	typ, ok := vim25MapType(name)
+	if !ok {
+		// See TestIssue945, in which case Go does not resolve the namespace and name == "ns1:TraversalSpec"
+		// Without this hack, the SelectSet would be all nil's
```
Comment (anfernee) [inline simulator/simulator.go]: need to take a closer look. feels hacky. seems like we are trimming only `vim25:` as namespace here: 

https://github.com/vmware/govmomi/blob/5072cda664c79ada30834d171d2ed1f76317d3b2/vim25/types/registry.go#L38

Comment (dougm) [reply simulator/simulator.go]: It is a hack, like the comment says.  We've only hit this with pyvsphere (not pyvmomi or other clients).  I don't want to spend time messing with our fork of the xml package at the moment.

Comment (anfernee) [reply simulator/simulator.go]: not the xml fork. just `TypeFunc` in `types/registry.go`. If you replace the `TrimPrefix` with `SplitN`, it should work as well. or maybe we can have a `TypeFunc` that only trims vim25, and anther that trims all namespaces.

Comment (dougm) [reply simulator/simulator.go]: Yeah I didn't want to put the hack in the types package.  The vim25 check there is valid.  But the hack above we shouldn't need at all, as the xml package should be properly resolving the namespace alias.

Comment (anfernee) [reply simulator/simulator.go]: oh, i see, there is this: `<SOAP-ENV:Body xmlns:ns1="urn:vim25">` missed that.

**Pull request merged**